mne_connectivity/spectral.py

```diff
@@ -916,19 +926,63 @@ def spectral_connectivity(data, method='coh', indices=None, sfreq=2 * np.pi,
                                 this_con_flat.shape[1:],
                                 dtype=this_con_flat.dtype)
             this_con[indices_use] = this_con_flat
+
+            con.append(this_con)
+    else:
+        # keep all arrays as all-to-all connectivity with nans
+        logger.info('    assembling connectivity matrix for subset')
+        con_flat = con
+        con = list()
+        for this_con_flat in con_flat:
+            this_con = np.zeros((n_signals, n_signals) +
+                                this_con_flat.shape[1:],
+                                dtype=this_con_flat.dtype)
+            this_con[:] = np.nan
+            this_con[indices_use] = this_con_flat
```

Comment:
@larsoner do you think this could be messing w/ the RAM since it might scale up the amount of RAM needed when the user only selects a few indices?

Reply:
Yeah it could. So maybe we do need to efficiently store the connectivity. Maybe look into / think about `scipy.sparse` like @drammock mentioned?

Reply:
Yeah looks like scipy has a sparse graph representation that supports nans/infs 

https://docs.scipy.org/doc/scipy/reference/sparse.csgraph.html#module-scipy.sparse.csgraph

I'll have to do some reading to see if Xarray plays nicely with that, else this will just be an edge case in the implementation for now.